ddl/ddl_api.go

```diff
@@ -3474,10 +3471,72 @@ func (d *ddl) UnlockTables(ctx sessionctx.Context, unlockTables []model.TableLoc
 	return errors.Trace(err)
 }
 
+func checkLockInMemOrSysDB(ctx sessionctx.Context, dbLowerName string) error {
```

Comment:
I think this function is nothing to do with `Lock`?
IMHO, it should name as `throwErrIfInMemOrSysDB`.